python/benchmark/gen_data_distributed.py

```diff
@@ -232,6 +234,152 @@ def make_matrix_udf(iter: Iterable[pd.DataFrame]) -> Iterable[pd.DataFrame]:
         )
 
 
+class RegressionDataGen(DataGenBaseMeta):
+    """Generate regression dataset using a distributed version of sklearn.datasets.regression,
+    including features and labels.
+    """
+
+    def __init__(self, argv: List[Any]) -> None:
+        super().__init__()
+        self._parse_arguments(argv)
+
+    def _supported_extra_params(self) -> Dict[str, Any]:
+        params = inspect_default_params_from_func(
+            make_regression, ["n_samples", "n_features", "coef"]
+        )
+        # must replace the None to the correct type
+        params["effective_rank"] = int
+        params["random_state"] = int
+        return params
+
+    def gen_dataframe_and_meta(
+        self, spark: SparkSession
+    ) -> Tuple[DataFrame, List[str], np.ndarray]:
+        dtype = self.dtype
+        params = self.extra_params
+
+        if "random_state" not in params:
+            # for reproducible dataset.
+            params["random_state"] = 1
+
+        print(f"Passing {params} to make_regression")
+
+        rows = self.num_rows
+        cols = self.num_cols
+        assert self.args is not None
+        num_partitions = self.args.output_num_files
+
+        # Set num_partitions to Spark's default if output_num_files is not provided.
+        if num_partitions is None:
+            num_partitions = spark.sparkContext.defaultParallelism
+
+        # Retrieve input params or set to defaults.
+        seed = params["random_state"]
+        generator = np.random.RandomState(seed)
+        bias = params.get("bias", 0.0)
+        noise = params.get("noise", 0.0)
+        shuffle = params.get("shuffle", True)
+        effective_rank = params.get("effective_rank", None)
+        n_informative = params.get("n_informative", 10)
+        n_targets = params.get("n_targets", 1)
+
+        # Input set is either well conditioned (default) or has a low rank fat tail singular profile (see LowRankMatrixDataGen).
+        # Output is generated by applying a (potentially biased) random linear regression model to the input, with n_informative
+        # nonzero regressors and some gaussian centered noise with adjustable scale.
+        # Code adapted from sklearn.datasets.make_regression().
```

Comment:
Need to specify "Comments (line 286-288) come from sklearn.datasets.make_regression()." as well.

Reply:
Not sure what you mean - as in, I should specify which exact code lines were adapted from sklearn?